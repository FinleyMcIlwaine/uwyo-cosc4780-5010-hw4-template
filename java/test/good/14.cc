// Macros to shut up the C++ warnings
#define printInt(x)
#define printDouble(x)
#define printString(x)
#define readInt() 0
#define readDouble() 0
#define readString() 0

// This program approximates pi and then checks that the approximation is within
// an acceptable bound

double piApprox(int iterations)
{
    double pi = 0.0;
    double factor = 1;
    int i = 2;
    while (i < iterations)
    {
        pi = pi + factor / (i - 1.0);
        i = i + 2;
        factor = factor * (0 - 1);
    }
    return 4 * pi;
}

int main()
{
    printString("How many iterations: ");
    int iterations = readInt();
    double pi = piApprox(iterations);
    // printDouble(pi);
    if (pi < 3.20 && pi > 3.10)
    {
        printString("pi approximated!\n");
    }
    else
    {
        printString("pi not approximated :,(\n");
    }
}