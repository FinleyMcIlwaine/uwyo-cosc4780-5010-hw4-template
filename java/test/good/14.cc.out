How many iterations: pi approximated!
